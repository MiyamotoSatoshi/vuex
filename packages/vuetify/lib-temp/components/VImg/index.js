import VImg from './VImg';
export { VImg };
export default VImg;
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi8uLi9zcmMvY29tcG9uZW50cy9WSW1nL2luZGV4LnRzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQUFBLE9BQU8sSUFBSSxNQUFNLFFBQVEsQ0FBQTtBQUV6QixPQUFPLEVBQUUsSUFBSSxFQUFFLENBQUE7QUFDZixlQUFlLElBQUksQ0FBQSIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCBWSW1nIGZyb20gJy4vVkltZydcblxuZXhwb3J0IHsgVkltZyB9XG5leHBvcnQgZGVmYXVsdCBWSW1nXG4iXX0=