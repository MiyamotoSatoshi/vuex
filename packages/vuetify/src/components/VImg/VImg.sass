@import './_variables.scss'

.v-img
  --v-theme-overlay-multiplier: 3
  z-index: 0

.v-img__img,
.v-img__picture,
.v-img__placeholder,
.v-img__error
  z-index: -1
  position: absolute
  top: 0
  left: 0
  width: 100%
  height: 100%

.v-img__img
  &--preload
    filter: $img-preload-filter

  &--contain
    background-size: contain

  &--cover
    background-size: cover
