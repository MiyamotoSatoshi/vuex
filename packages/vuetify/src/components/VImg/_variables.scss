// Defaults
$img-preload-filter: blur(4px) !default;
