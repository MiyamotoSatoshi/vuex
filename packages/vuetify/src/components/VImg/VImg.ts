// Styles
import './VImg.sass'

// Directives
import intersect from '../../directives/intersect'

// Types
import { PropValidator, PropType } from 'vue/types/options'
import {
  VNode,
  VNodeDirective,
} from 'vue'

// Components
import VResponsive from '../VResponsive'

// Mixins
import Themeable from '../../mixins/themeable'

// Utils
import mergeData from '../../util/mergeData'
import mixins from '../../util/mixins'
import { consoleError, consoleWarn } from '../../util/console'

// not intended for public use, this is passed in by vuetify-loader
export interface srcObject {
  src: string
  srcset?: string
  lazySrc: string
  aspect: number
}

const baseMixins = mixins(
  VResponsive,
  Themeable,
)

/* @vue/component */
export default baseMixins.extend({
  name: 'v-img',

  directives: { intersect },

  props: {
    alt: String,
    contain: Boolean,
    eager: Boolean,
    gradient: String,
    lazySrc: String,
    options: {
      type: Object,
      // For more information on types, navigate to:
      // https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API
      default: () => ({
        root: undefined,
        rootMargin: undefined,
        threshold: undefined,
      }),
    } as PropValidator<IntersectionObserverInit>,
    position: {
      type: String,
      default: 'center center',
    },
    sizes: String,
    src: {
      type: [String, Object] as PropType<string | srcObject>,
      default: '',
    },
    srcset: String,
    transition: {
      type: [Boolean, String],
      default: 'fade-transition',
    },
  },

  data () {
    return {
      currentSrc: '', // Set from srcset
      image: null as HTMLImageElement | null,
      isLoading: true,
      calculatedAspectRatio: undefined as number | undefined,
      naturalWidth: undefined as number | undefined,
    }
  },

  computed: {
    computedAspectRatio (): number | undefined {
      const aspectRatio = Number(this.normalisedSrc.aspect || this.calculatedAspectRatio)

      return !isNaN(aspectRatio) ? (1 / aspectRatio) : undefined
    },
    hasIntersect () {
      return (
        typeof window !== 'undefined' &&
        'IntersectionObserver' in window
      )
    },
    normalisedSrc (): srcObject {
      return typeof this.src === 'string'
        ? {
          src: this.src,
          srcset: this.srcset,
          lazySrc: this.lazySrc,
          aspect: Number(this.aspectRatio || 0),
        } : {
          src: this.src.src,
          srcset: this.srcset || this.src.srcset,
          lazySrc: this.lazySrc || this.src.lazySrc,
          aspect: Number(this.aspectRatio || this.src.aspect),
        }
    },
    __cachedImage (): VNode | [] {
      if (!(this.normalisedSrc.src || this.normalisedSrc.lazySrc)) return []

      const backgroundImage: string[] = []
      const src = this.isLoading ? this.normalisedSrc.lazySrc : this.currentSrc

      if (this.gradient) backgroundImage.push(`linear-gradient(${this.gradient})`)
      if (src) backgroundImage.push(`url("${src}")`)

      const image = this.$createElement('div', {
        staticClass: 'v-image__image',
        class: {
          'v-image__image--preload': this.isLoading,
          'v-image__image--contain': this.contain,
          'v-image__image--cover': !this.contain,
        },
        style: {
          backgroundImage: backgroundImage.join(', '),
          backgroundPosition: this.position,
        },
        key: +this.isLoading,
      })

      /* istanbul ignore if */
      if (!this.transition) return image

      return this.$createElement('transition', {
        attrs: {
          name: this.transition,
          mode: 'in-out',
        },
      }, [image])
    },
  },

  watch: {
    src () {
      // Force re-init when src changes
      if (!this.isLoading) this.init(undefined, undefined, true)
      else this.loadImage()
    },
    '$vuetify.breakpoint.width': 'getSrc',
  },

  mounted () {
    this.init()
  },

  methods: {
    init (
      entries?: IntersectionObserverEntry[],
      observer?: IntersectionObserver,
      isIntersecting?: boolean
    ) {
      // If the current browser supports the intersection
      // observer api, the image is not observable, and
      // the eager prop isn't being used, do not load
      if (
        this.hasIntersect &&
        !isIntersecting &&
        !this.eager
      ) return

      if (this.normalisedSrc.lazySrc) {
        const lazyImg = new Image()
        lazyImg.src = this.normalisedSrc.lazySrc
        this.pollForSize(lazyImg, null)
      }
      /* istanbul ignore else */
      if (this.normalisedSrc.src) this.loadImage()
    },
    onLoad () {
      this.getSrc()
      this.isLoading = false
      this.$emit('load', this.src)
    },
    onError () {
      consoleError(
        `Image load failed\n\n` +
        `src: ${this.normalisedSrc.src}`,
        this
      )
      this.$emit('error', this.src)
    },
    genContent (): VNode | undefined {
      return this.computedAspectRatio != null
        ? VResponsive.options.methods.genContent.call(this)
        : undefined
    },
    getSrc () {
      /* istanbul ignore else */
      if (this.image) this.currentSrc = this.image.currentSrc || this.image.src
    },
    loadImage () {
      const image = new Image()
      this.image = image

      image.onload = () => {
        /* istanbul ignore if */
        if (image.decode) {
          image.decode().catch((err: DOMException) => {
            consoleWarn(
              `Failed to decode image, trying to render anyway\n\n` +
              `src: ${this.normalisedSrc.src}` +
              (err.message ? `\nOriginal error: ${err.message}` : ''),
              this
            )
          }).then(this.onLoad)
        } else {
          this.onLoad()
        }
      }
      image.onerror = this.onError

      image.src = this.normalisedSrc.src
      this.sizes && (image.sizes = this.sizes)
      this.normalisedSrc.srcset && (image.srcset = this.normalisedSrc.srcset)

      this.aspectRatio || this.pollForSize(image)
      this.getSrc()
    },
    pollForSize (img: HTMLImageElement, timeout: number | null = 100) {
      const poll = () => {
        const { naturalHeight, naturalWidth } = img

        if (naturalHeight || naturalWidth) {
          this.naturalWidth = naturalWidth
          this.calculatedAspectRatio = naturalWidth / naturalHeight
        } else {
          timeout != null && setTimeout(poll, timeout)
        }
      }

      poll()
    },
    __genPlaceholder (): VNode | undefined {
      if (!this.$slots.placeholder) return undefined

      const placeholder = this.isLoading
        ? [this.$createElement('div', {
          staticClass: 'v-image__placeholder',
        }, this.$slots.placeholder)]
        : []

      if (!this.transition) return placeholder[0]

      return this.$createElement('transition', {
        props: {
          appear: true,
          name: this.transition,
        },
      }, placeholder)
    },
  },

  render (h): VNode {
    const directives = []
    const node = VResponsive.options.render.call(this, h)

    // Only load intersect directive if it
    // will work in the current browser.
    if (this.hasIntersect) {
      directives.push({
        name: 'intersect',
        options: this.options,
        modifiers: { once: true },
        value: this.init,
      } as VNodeDirective)
    }

    const data = mergeData(node.data!, {
      staticClass: 'v-image',
      attrs: {
        role: this.alt ? 'img' : undefined,
        'aria-label': this.alt,
      },
      directives,
    })

    return h(node.tag, data, [
      this.__cachedSizer,
      this.__cachedImage,
      this.__genPlaceholder(),
      this.genContent(),
    ])
  },
})
