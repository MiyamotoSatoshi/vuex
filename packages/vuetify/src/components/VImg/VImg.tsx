import './VImg.sass'

// Vue
import {
  computed,
  defineComponent,
  h,
  nextTick,
  onBeforeMount,
  ref,
  vShow,
  watch,
  withDirectives,
} from 'vue'
import type { PropType } from 'vue'

// Components
import { VResponsive } from '../VResponsive'

// Directives
import intersect from '@/directives/intersect'
import type { ObserveDirectiveBinding } from '@/directives/intersect'

// Utils
import makeProps from '@/util/makeProps'
import { useDirective } from '@/util/useDirective'
import { maybeTransition } from '@/util'

// not intended for public use, this is passed in by vuetify-loader
export interface srcObject {
  src?: string
  srcset?: string
  lazySrc?: string
  aspect: number
}

const hasIntersect = typeof window !== 'undefined' && 'IntersectionObserver' in window

export default defineComponent({
  name: 'VImg',

  props: makeProps({
    aspectRatio: [String, Number],
    alt: String,
    cover: Boolean,
    eager: Boolean,
    lazySrc: String,
    options: {
      type: Object as PropType<IntersectionObserverInit>,
      // For more information on types, navigate to:
      // https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API
      default: () => ({
        root: undefined,
        rootMargin: undefined,
        threshold: undefined,
      }),
    },
    position: {
      type: String,
      default: 'center center',
    },
    sizes: String,
    src: {
      type: [String, Object] as PropType<string | srcObject>,
      default: '',
    },
    srcset: String,
    transition: {
      type: [Boolean, String] as PropType<string | false>,
      default: 'fade-transition',
      validator: val => val !== true,
    },
  }),

  emits: ['loadstart', 'load', 'error'],

  setup (props, { emit, expose, slots }) {
    const currentSrc = ref('') // Set from srcset
    const image = ref<HTMLImageElement>()
    const state = ref<'idle' | 'loading' | 'loaded' | 'error'>('idle')
    const naturalWidth = ref<number>()
    const naturalHeight = ref<number>()

    expose({
      currentSrc,
      image,
      state,
      naturalWidth,
      naturalHeight,
    })

    const normalisedSrc = computed<srcObject>(() => {
      return props.src && typeof props.src === 'object'
        ? {
          src: props.src.src,
          srcset: props.srcset || props.src.srcset,
          lazySrc: props.lazySrc || props.src.lazySrc,
          aspect: Number(props.aspectRatio || props.src.aspect),
        } : {
          src: props.src,
          srcset: props.srcset,
          lazySrc: props.lazySrc,
          aspect: Number(props.aspectRatio || 0),
        }
    })
    const aspectRatio = computed(() => {
      return normalisedSrc.value.aspect || naturalWidth.value! / naturalHeight.value! || 0
    })

    watch(() => props.src, () => {
      init(state.value !== 'idle')
    })
    // TODO: getSrc when window width changes

    onBeforeMount(() => init())

    function init (isIntersecting?: boolean) {
      // If the current browser supports the intersection
      // observer api, the image is not observable, and
      // the eager prop isn't being used, do not load
      if (
        hasIntersect &&
        !isIntersecting &&
        !props.eager
      ) return

      state.value = 'loading'
      nextTick(() => {
        emit('loadstart', image.value?.currentSrc || props.src)
        if (!aspectRatio.value) pollForSize(image.value!)
        getSrc()
      })

      if (normalisedSrc.value.lazySrc) {
        const lazyImg = new Image()
        lazyImg.src = normalisedSrc.value.lazySrc
        pollForSize(lazyImg, null)
      }
    }

    function onLoad () {
      getSrc()
      state.value = 'loaded'
      emit('load', image.value?.currentSrc || props.src)
    }

    function onError () {
      state.value = 'error'
      emit('error', image.value?.currentSrc || props.src)
    }

    function getSrc () {
      const img = image.value
      if (img) currentSrc.value = img.currentSrc || img.src
    }

    function pollForSize (img: HTMLImageElement, timeout: number | null = 100) {
      const poll = () => {
        const { naturalHeight: imgHeight, naturalWidth: imgWidth } = img

        if (imgHeight || imgWidth) {
          naturalWidth.value = imgWidth
          naturalHeight.value = imgHeight
        } else if (!img.complete && state.value === 'loading' && timeout != null) {
          setTimeout(poll, timeout)
        } else if (img.currentSrc.endsWith('.svg') || img.currentSrc.startsWith('data:image/svg+xml')) {
          naturalWidth.value = 1
          naturalHeight.value = 1
        }
      }

      poll()
    }

    const containClasses = computed(() => ({
      'v-img__img--cover': props.cover,
      'v-img__img--contain': !props.cover,
    }))

    const __image = computed(() => {
      if (!normalisedSrc.value.src || state.value === 'idle') return

      const img = h('img', {
        class: ['v-img__img', containClasses.value],
        src: normalisedSrc.value.src,
        srcset: normalisedSrc.value.srcset,
        sizes: props.sizes,
        ref: image,
        onLoad,
        onError,
      })

      const sources = slots.sources?.()

      return maybeTransition(
        props,
        { appear: true },
        withDirectives(
          sources
            ? <picture class="v-img__picture">{sources}{img}</picture>
            : img,
          [[vShow, state.value === 'loaded']]
        )
      )
    })

    const __preloadImage = computed(() => {
      return maybeTransition(
        props,
        {},
        normalisedSrc.value.lazySrc && state.value !== 'loaded' ? (
          <img
            class={['v-img__img', 'v-img__img--preload', containClasses.value]}
            src={ normalisedSrc.value.lazySrc }
            alt=""
          />
        ) : undefined
      )
    })

    const __placeholder = computed(() => {
      if (!slots.placeholder) return

      const placeholder = state.value === 'loading' || (state.value === 'error' && !slots.error)
        ? <div class="v-img__placeholder">{ slots.placeholder() }</div>
        : undefined

      return maybeTransition(props, { appear: true }, placeholder)
    })

    const __error = computed(() => {
      if (!slots.error) return

      const error = state.value === 'error'
        ? <div class="v-img__error">{ slots.error() }</div>
        : undefined

      return maybeTransition(props, { appear: true }, error)
    })

    return () => withDirectives(
      <VResponsive
        class="v-img"
        aspectRatio={ aspectRatio.value }
        aria-label={ props.alt }
        role={ props.alt ? 'img' : undefined }
        v-slots={{
          additional: () => [__image.value, __preloadImage.value, __placeholder.value, __error.value],
          default: slots.default,
        }}
      />,
      [useDirective<ObserveDirectiveBinding>(intersect, {
        value: {
          handler: init,
          options: props.options,
        },
        modifiers: { once: true },
      })]
    )
  },
})
