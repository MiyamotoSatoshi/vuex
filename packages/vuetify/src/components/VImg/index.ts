import VImg from './VImg'

export { VImg }
