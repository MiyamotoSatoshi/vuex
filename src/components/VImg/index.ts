import VImg from './VImg'

/* istanbul ignore next */
VImg.install = function install (Vue) {
  Vue.component(VImg.options.name, VImg)
}

export { VImg }
export default VImg
