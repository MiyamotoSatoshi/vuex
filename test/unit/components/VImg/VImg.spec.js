import { test } from '@/test'
import VImg from '@/components/VImg'

test('VImg', ({ mount }) => {
  const LOAD_FAILURE_SRC = 'LOAD_FAILURE_SRC'
  const LOAD_SUCCESS_SRC = 'LOAD_SUCCESS_SRC'

  beforeAll(() => {
    jest.useFakeTimers()
    Object.defineProperty(global.Image.prototype, 'src', {
      set (src) {
        this._currentSrc = src
        if (src === LOAD_FAILURE_SRC) {
          setTimeout(() => this.onerror && this.onerror(new Error('mocked error')))
        } else {
          setTimeout(() => {
            this._naturalWidth = 1600
            this._naturalHeight = 900
            this.onload && this.onload()
          })
        }
      }
    })
    Object.defineProperty(global.Image.prototype, 'currentSrc', {
      get () {
        return this._currentSrc
      }
    })
    Object.defineProperty(global.Image.prototype, 'naturalWidth', {
      get () { return this._naturalWidth }
    })
    Object.defineProperty(global.Image.prototype, 'naturalHeight', {
      get () { return this._naturalHeight }
    })
  })

  afterAll(() => {
    jest.useRealTimers()
  })

  it('should load', async () => {
    const wrapper = mount(VImg, {
      propsData: { src: LOAD_SUCCESS_SRC }
    })

    expect(wrapper.html()).toMatchSnapshot()

    jest.runOnlyPendingTimers()
    await wrapper.vm.$nextTick()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should display placeholders', async () => {
    const wrapper = mount(VImg, {
      propsData: {
        src: 'full_src',
        lazySrc: 'lazy_src'
      },
      slots: {
        placeholder: { render: h => h('div', ['loading...'])}
      }
    })

    expect(wrapper.html()).toMatchSnapshot()

    jest.runOnlyPendingTimers()
    await wrapper.vm.$nextTick()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should emit errors', () => {
    const wrapper = mount(VImg, {
      propsData: { src: LOAD_FAILURE_SRC }
    })

    const error = jest.fn()
    wrapper.vm.$on('error', error)

    jest.runOnlyPendingTimers()

    expect(error).toHaveBeenCalledTimes(1)
    expect(error).toHaveBeenCalledWith(LOAD_FAILURE_SRC)
    expect('Image load failed').toHaveBeenWarned()
  })

  it('should have aria attributes', async () => {
    const wrapper = mount(VImg, {
      propsData: {
        src: LOAD_SUCCESS_SRC,
        alt: 'this is not a decorative image'
      }
    })

    jest.runOnlyPendingTimers()
    await wrapper.vm.$nextTick()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should use vuetify-loader data', async () => {
    const wrapper = mount(VImg, {
      propsData: {
        src: {
          src: LOAD_SUCCESS_SRC,
          lazySrc: 'lazySrc_auto',
          aspect: 1
        }
      }
    })

    jest.runOnlyPendingTimers()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should override vuetify-loader values', async () => {
    const wrapper = mount(VImg, {
      propsData: {
        src: {
          src: LOAD_SUCCESS_SRC,
          lazySrc: 'lazySrc_auto',
          aspect: 1
        },
        lazySrc: 'lazySrc_manual',
        aspectRatio: 2
      }
    })

    jest.runOnlyPendingTimers()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should update src', async () => {
    const wrapper = mount(VImg, {
      propsData: {
        src: LOAD_SUCCESS_SRC
      }
    })

    jest.runOnlyPendingTimers()
    await wrapper.vm.$nextTick()

    expect(wrapper.html()).toMatchSnapshot()

    wrapper.setProps({ src: LOAD_SUCCESS_SRC + 1 })

    jest.runOnlyPendingTimers()
    await wrapper.vm.$nextTick()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should update src while still loading', async () => {
    const wrapper = mount(VImg, {
      propsData: {
        src: LOAD_SUCCESS_SRC
      }
    })

    expect(wrapper.html()).toMatchSnapshot()

    wrapper.setProps({ src: LOAD_SUCCESS_SRC + 1 })

    jest.runOnlyPendingTimers()
    await wrapper.vm.$nextTick()

    expect(wrapper.html()).toMatchSnapshot()
  })
})
