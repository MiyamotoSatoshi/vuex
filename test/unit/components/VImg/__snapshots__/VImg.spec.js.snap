// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VImg should display placeholders 1`] = `

<div class="v-responsive v-image">
  <div class="v-image__image v-image__image--preload v-image__image--cover"
       style="background-image: url(lazy_src); background-position: center center;"
  >
  </div>
  <div class="v-image__placeholder">
    <div>
      loading...
    </div>
  </div>
  <div class="v-responsive__content">
  </div>
</div>

`;

exports[`VImg should display placeholders 2`] = `

<div class="v-responsive v-image">
  <div class="v-responsive__sizer"
       style="padding-bottom: 56.25%;"
  >
  </div>
  <div class="v-image__image v-image__image--preload v-image__image--cover"
       style="background-image: url(lazy_src); background-position: center center;"
  >
  </div>
  <div class="v-image__image v-image__image--cover fade-transition-enter fade-transition-enter-active"
       style="background-image: url(full_src); background-position: center center;"
  >
  </div>
  <div class="v-image__placeholder fade-transition-leave fade-transition-leave-active">
    <div>
      loading...
    </div>
  </div>
  <div class="v-responsive__content">
  </div>
</div>

`;

exports[`VImg should have aria attributes 1`] = `

<div role="img"
     aria-label="this is not a decorative image"
     class="v-responsive v-image"
>
  <div class="v-responsive__sizer"
       style="padding-bottom: 56.25%;"
  >
  </div>
  <div class="v-image__image v-image__image--preload v-image__image--cover"
       style="background-position: center center;"
  >
  </div>
  <div class="v-image__image v-image__image--cover fade-transition-enter fade-transition-enter-active"
       style="background-image: url(LOAD_SUCCESS_SRC); background-position: center center;"
  >
  </div>
  <div class="v-responsive__content">
  </div>
</div>

`;

exports[`VImg should load 1`] = `

<div class="v-responsive v-image">
  <div class="v-image__image v-image__image--preload v-image__image--cover"
       style="background-position: center center;"
  >
  </div>
  <div class="v-responsive__content">
  </div>
</div>

`;

exports[`VImg should load 2`] = `

<div class="v-responsive v-image">
  <div class="v-responsive__sizer"
       style="padding-bottom: 56.25%;"
  >
  </div>
  <div class="v-image__image v-image__image--preload v-image__image--cover"
       style="background-position: center center;"
  >
  </div>
  <div class="v-image__image v-image__image--cover fade-transition-enter fade-transition-enter-active"
       style="background-image: url(LOAD_SUCCESS_SRC); background-position: center center;"
  >
  </div>
  <div class="v-responsive__content">
  </div>
</div>

`;

exports[`VImg should override vuetify-loader values 1`] = `

<div class="v-responsive v-image">
  <div class="v-responsive__sizer"
       style="padding-bottom: 50%;"
  >
  </div>
  <div class="v-image__image v-image__image--preload v-image__image--cover"
       style="background-image: url(lazySrc_manual); background-position: center center;"
  >
  </div>
  <div class="v-responsive__content">
  </div>
</div>

`;

exports[`VImg should update src 1`] = `

<div class="v-responsive v-image">
  <div class="v-responsive__sizer"
       style="padding-bottom: 56.25%;"
  >
  </div>
  <div class="v-image__image v-image__image--preload v-image__image--cover"
       style="background-position: center center;"
  >
  </div>
  <div class="v-image__image v-image__image--cover fade-transition-enter fade-transition-enter-active"
       style="background-image: url(LOAD_SUCCESS_SRC); background-position: center center;"
  >
  </div>
  <div class="v-responsive__content">
  </div>
</div>

`;

exports[`VImg should update src 2`] = `

<div class="v-responsive v-image">
  <div class="v-responsive__sizer"
       style="padding-bottom: 56.25%;"
  >
  </div>
  <div class="v-image__image v-image__image--preload v-image__image--cover"
       style="background-position: center center;"
  >
  </div>
  <div class="v-image__image v-image__image--cover fade-transition-enter fade-transition-enter-active"
       style="background-image: url(LOAD_SUCCESS_SRC1); background-position: center center;"
  >
  </div>
  <div class="v-responsive__content">
  </div>
</div>

`;

exports[`VImg should update src while still loading 1`] = `

<div class="v-responsive v-image">
  <div class="v-image__image v-image__image--preload v-image__image--cover"
       style="background-position: center center;"
  >
  </div>
  <div class="v-responsive__content">
  </div>
</div>

`;

exports[`VImg should update src while still loading 2`] = `

<div class="v-responsive v-image">
  <div class="v-responsive__sizer"
       style="padding-bottom: 56.25%;"
  >
  </div>
  <div class="v-image__image v-image__image--preload v-image__image--cover"
       style="background-position: center center;"
  >
  </div>
  <div class="v-image__image v-image__image--cover fade-transition-enter fade-transition-enter-active"
       style="background-image: url(LOAD_SUCCESS_SRC1); background-position: center center;"
  >
  </div>
  <div class="v-responsive__content">
  </div>
</div>

`;

exports[`VImg should use vuetify-loader data 1`] = `

<div class="v-responsive v-image">
  <div class="v-responsive__sizer"
       style="padding-bottom: 100%;"
  >
  </div>
  <div class="v-image__image v-image__image--preload v-image__image--cover"
       style="background-image: url(lazySrc_auto); background-position: center center;"
  >
  </div>
  <div class="v-responsive__content">
  </div>
</div>

`;
